### DISCUSSION TITLE
Running Python in the query transformations

### DISCUSSION THREAD

**Discussion by dianale94** [ToolJet/ToolJet#8140] [Q&A]
Hi, I'm grabbing data from a Dynamo table and querying by a userId. I now want to do some transformations on the row retrieved for the userId with code that looks something like the following:
```
import json

transformed = []
for row in data:
	json_data = json.loads(row['data'])
        // Do some indexing and transformations into the json_data
       // ...
	transformed.append({
		'user_id': row['user_id'],
		//add transformed variables here
	})
return transformed
```

If I attempt to run that code (or if I transform the for-loop into a list comprehension), I get `IndentationError: unexpected indent`.
 
Let me know if I'm doing something obviously wrong, but also a few questions:

1. Does Python in Tooljet's Transformations allow for imports like this? My guess is no, as shown by the IndentationError, but it also seems like the json library is not supported out of the box?
2. Is there another way to transform my data? The data is a valid json string and I'd like to be able to manipulate it without needing to do just brute string manipulation.

**Comment by ShazanRizvi** [comment]
Hey @dianale94 
You can [import packages in ToolJet using RunPy](https://docs.tooljet.com/docs/how-to/import-external-libraries-using-runpy/). For transformations you can also use [Javascript transformations](https://docs.tooljet.com/docs/tutorial/transformations/#transform-using-javascript).
You can also join our slack community to get help from our team and our community members. Here is the invite: https://tooljet.com/slack.
